selfdrive/car/honda/carstate.py

```diff
@@ -263,6 +262,21 @@ def update(self, cp, cp_cam, cp_body):
     ret.cruiseState.enabled = cp.vl["POWERTRAIN_DATA"]["ACC_STATUS"] != 0
     ret.cruiseState.available = bool(cp.vl[self.main_on_sig_msg]["MAIN_ON"])
 
+    steer_status = self.steer_status_values[cp.vl["STEER_STATUS"]["STEER_STATUS"]]
+    # LOW_SPEED_LOCKOUT is not worth a warning on most cars
+    # NO_TORQUE_ALERT_2 can be caused by bump or steering nudge from driver
+    ret.steerFaultPermanent = steer_status not in ("NORMAL", "NO_TORQUE_ALERT_1", "NO_TORQUE_ALERT_2", "LOW_SPEED_LOCKOUT", "TMP_FAULT")
+    if self.CP.minSteerSpeed > 0.:
+      # TODO: make generic for all makes and clean up
+      # Check for steer mismatch when engaged and moving before setting a temporary fault. Allow for a small buffer as to not flash the alert when engaging.
+      self.steer_mismatch_cnt = self.steer_mismatch_cnt + 1 if ret.cruiseState.enabled and \
+                                not bool(cp.vl["STEER_STATUS"]["STEER_CONTROL_ACTIVE"]) and not ret.standstill else 0
+      steer_mismatch = True if self.steer_mismatch_cnt > MAX_STEPS_STEER_MISMATCH else False
```

Comment:
What does this check exactly?

Reply:
It checks to see how many timesteps it takes for our STEER_TORQUE_REQUEST bit to make it to the radar > out of the radar if conditions are good > read by the EPS > if the EPS accepts the request and its conditions are good > EPS sends the status bit active on the feedback frame > this gets read back into carstate. The number of timesteps that it takes for all of this to happen isn't always consistent.

This becomes important when we engage and the radar IS allowing steering. It ensures that we don't flash the steering temporarily unavailable alert at the driver during the brief moment that we've engaged and sent our torque request to the radar but the EPS hasn't yet indicated that control is active.

Reply:
Maybe I missed it, but do you know the conditions that the radar will not allow steering besides the minimum steering speed? If we know if any consistent cases, perhaps we can find a signal from the radar that describes a forwarding fault so we don't need this complex logic around frame thresholds, checking standstill, etc

Reply:
So far it appears to be: speed, braking, windshield wiper mode constant (high or low speed), and steer status (faulted, too great of steer angle vs speed, etc) but it's possible there are cases that I haven't seen. I've looked before for feedback from the radar but hadn't found anything that correlated exactly; I'll spend more time researching this.

Reply:
Desktop Cabana has a great tool to find similar bits, might be helpful here. I also have a script I use where I find a heuristic for the signal I'm looking for and then set `bit_to_find`. For this case, you can find an example of the radar not forwarding the steering message, then set the bit to find to 1 when the steer status from the EPS is not active but OP is sending steer.

Cabana is useful to find similar bits to an existing signal, but I use my script for anything else, especially if I don't have anything to go off of https://github.com/commaai/openpilot/pull/26683